Doc/whatsnew/3.12.rst

```diff
@@ -236,6 +236,10 @@ New Features
   an additional metaclass argument.
   (Contributed by Wenzel Jakob in :gh:`93012`.)
 
+* Add new function :c:func:`PyFunction_SetVectorcall` to the C API
+  which sets the vectorcall field of a given PyFunctionObject
```

Comment:
You may explain the purpose of the function here, like mention Cinder JIT and Pyjion.

Reply:
I'd rather not have a "purpose". We should state clearly what the function does, not bless some particular use case.

@adphrost
Could you add a warning that the new function pointer must have **exactly** the same behavior as the unaltered function.
I am a little concerned that C extensions will abuse this, and we will have deal with the bug reports.